packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,528 @@ const double _kHandleSize = 22.0;
+    // The x, y parameters have the top left of the node's box as the origin.
+    RenderBox child = lastChild;
+    int i = childCount;
+    while (child != null) {
+      i--;
+      final FlexParentData childParentData = child.parentData as FlexParentData;
+
+      // Don't hit test children that haven't been painted.
+      if (!_shouldPaintChild(child, i)) {
+        child = childParentData.previousSibling;
+        continue;
+      }
+
+      final bool isHit = result.addWithPaintOffset(
+        offset: childParentData.offset,
+        position: position,
+        hitTest: (BoxHitTestResult result, Offset transformed) {
+          assert(transformed == position - childParentData.offset);
+          return child.hitTest(result, position: transformed);
+        },
+      );
+      if (isHit)
+        return true;
+      child = childParentData.previousSibling;
+    }
+    return false;
+  }
+}
+
+// When the overflow menu is open, it tries to align its right edge to the right
+// edge of the closed menu. This widget handles this effect by measuring and
+// maintaining the width of the closed menu and aligning the child to the right.
+class _TextSelectionToolbarContainer extends SingleChildRenderObjectWidget {
+  const _TextSelectionToolbarContainer({
+    Key key,
+    @required Widget child,
+    @required this.menuChanged,
+    @required this.overflowOpen,
+  }) : assert(child != null),
+       assert(overflowOpen != null),
+       super(key: key, child: child);
+
+  final bool menuChanged;
+  final bool overflowOpen;
+
+  @override
+  _TextSelectionToolbarContainerRenderBox createRenderObject(BuildContext context) {
+    return _TextSelectionToolbarContainerRenderBox(overflowOpen: overflowOpen);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _TextSelectionToolbarContainerRenderBox renderObject) {
+    renderObject
+      ..overflowOpen = overflowOpen;
+
+    if (menuChanged) {
+      renderObject.onMenuChanged();
+    }
+  }
+}
+
+class _TextSelectionToolbarContainerRenderBox extends RenderProxyBox {
+  _TextSelectionToolbarContainerRenderBox({
+    @required bool overflowOpen,
+  }) : assert(overflowOpen != null),
+       _overflowOpen = overflowOpen,
+       super();
+
+  // The width of the menu when it was closed. This is used to achieve the
+  // behavior where the open menu aligns its right edge to the closed menu's
+  // right edge.
+  double _closedWidth;
+
+  bool _overflowOpen;
+  bool get overflowOpen => _overflowOpen;
+  set overflowOpen(bool value) {
+    if (value == overflowOpen) {
+      return;
+    }
+    _overflowOpen = value;
+    markNeedsLayout();
+  }
+
+  // If the contents of the menu changed, reset the tracked width.
+  void onMenuChanged() {
+    _closedWidth = null;
+    markNeedsLayout();
+  }
+
+  @override
+  void performLayout() {
+    child.layout(constraints.loosen(), parentUsesSize: true);
+
+    if (!overflowOpen && _closedWidth == null) {
```

Comment:
It matters because the size is animated.  In the frame that `overflowOpen` is first toggled to false, the menu is still at the "open" size, and the animation begins.  If I were to immediately set the size to the full open size, then its alignment would jump to the left, and then it would animate.

`_closedWidth` doesn't get reset to null after open/close, because the "closed" size of the menu doesn't change.  It only changes when the menu items change, and at that time `_closedWidth` is reset to null.

Reply:
Adding a comment about this.

Reply:
I also added a condition to handle the case where the menu is first built with the overflow menu open. In that case it will just render at the size of the open menu.